cli/commands/run/server/agent.controller.js

```diff
@@ -87,12 +102,47 @@ module.exports = class AgentController {
     });
   }
 
+  async EvaluateAlert(call, callback) {
+    const findings = [];
+    let status = "SUCCESS";
+
+    if (this.handleAlert) {
+      try {
+        const alertEvent = this.createAlertEventFromGrpcRequest(call.request);
+        const returnedFindings = await this.handleAlert(alertEvent);
+
+        assertFindings(returnedFindings)
+
+        findings.push(...returnedFindings);
+      } catch (e) {
+        console.log(
+            `${new Date().toISOString()}    evaluateAlert ${
+                call.request.hash
+            }`
+        );
+        console.log(e);
+        status = "ERROR";
+      }
+    }
+
+    callback(null, {
+      status,
+      findings,
+      metadata: {
+        timestamp: new Date().toISOString(),
+      },
+      private: isPrivateFindings(),
+    });
+  }
+
   async initializeAgentHandlers() {
     try {
       // getAgentHandlers will also call any initialize handler
       const agentHandlers = await this.getAgentHandlers();
```

Comment:
the `getAgentHandlers` method currently invokes `initialize` as well. this is intended for running the bot using the CLI `run` command. we should probably move the `initialize` call out of there (and somewhere else in the run flow so that its not called twice here)